### ISSUE TITLE
Make `amrex::ParallelFor` lambda bodies consistent

### ISSUE THREAD

**Issue by mirenradia** [GRTLCollaboration/GRTeclyn#8]
This could make the code in example level classes cleaner but it might not be worth it.

**Comment by KAClough** [comment]
I think just improving the naming and using the ParallelFor in a consistent way at different points in the code would make this sufficiently clear.

**Comment by KAClough** [comment]
We should also decide if we want to keep the compute naming, or generalise what we call the compute classes. I quite like the idea of making them more specific, e.g. compute_rhs(), compute_initial_data()...

**Comment by mirenradia** [comment]
We now have some inconsistencies as some former compute classes e.g. `TraceARemoval` use the `operator()` function instead.

> We should also decide if we want to keep the compute naming, or generalise what we call the compute classes. I quite like the idea of making them more specific, e.g. compute_rhs(), compute_initial_data()...

I'm a bit confused. Why do you want to "make them more specific"? Typically, the compute classes already have the thing they're calculating in their type name e.g. `CCZ4RHS` and when they're instantiated, the objects are usually called accordingly e.g. `ccz4rhs` so I don't see what is added by changing e.g.

```cpp
CCZ4RHS<...> ccz4rhs(...); // construct object
...
ccz4rhs.compute(...); // do computation in a loop
```
to
```cpp
CCZ4RHS<...> ccz4rhs(...); // construct object
...
ccz4rhs.compute_rhs(...); // do computation in a loop
```

I definitely agree we could be more consistent with the type names for the initial data compute classes (e.g. `BinaryBH` -> `BinaryBHInitialData` or `BinaryBHID`).

**Comment by KAClough** [comment]
Yes true. I think I just don't really like the name `compute()` but it is ok. Something more descriptive would be nice that makes it clear it is applied over all cells... but I can't think of anything that isn't quite long and potentially confusing in itself.

**Comment by mirenradia** [comment]
> Something more descriptive would be nice that makes it clear it is applied over all cells... but I can't think of anything that isn't quite long and potentially confusing in itself.

The `compute()` function is called each single cell. The `ParallelFor` loop over all cells is outside of it.

**Comment by KAClough** [comment]
Yes right, so the compute could be `compute_variables_in_cell()` or `compute_cell_values()`, then with a nicely formatted `ParallelFor` loop it should be clear to a newbie user what is happening.

**Comment by julianakwan** [comment]
We want to make the member function of the class in the lambda bodies consistent, e.g. sometimes these are named `compute`, sometimes invoked with parentheses or even `init_data`. 

I've renamed the issue because we decided that the wrapping of `Parallel_For` is not really necessary.

**Comment by mirenradia** [comment]
Completed in #138
